internal/provider/serverless_cluster_data_source.go

```diff
@@ -0,0 +1,352 @@
+						Computed:            true,
+					},
+					"display_name": schema.StringAttribute{
+						MarkdownDescription: "The display name of the region.",
+						Computed:            true,
+					},
+				},
+			},
+			"spending_limit": schema.SingleNestedAttribute{
+				MarkdownDescription: "The spending limit of the cluster.",
+				Computed:            true,
+				Attributes: map[string]schema.Attribute{
+					"monthly": schema.Int64Attribute{
+						MarkdownDescription: "Maximum monthly spending limit in USD cents.",
+						Computed:            true,
+					},
+				},
+			},
+			"automated_backup_policy": schema.SingleNestedAttribute{
+				MarkdownDescription: "The automated backup policy of the cluster.",
+				Computed:            true,
+				Attributes: map[string]schema.Attribute{
+					"start_time": schema.StringAttribute{
+						MarkdownDescription: "The time of day when the automated backup will start.",
+						Computed:            true,
+					},
+					"retention_days": schema.Int64Attribute{
+						MarkdownDescription: "The number of days to retain automated backups.",
+						Computed:            true,
+					},
+				},
+			},
+			"endpoints": schema.SingleNestedAttribute{
+				MarkdownDescription: "The endpoints for connecting to the cluster.",
+				Computed:            true,
+				Attributes: map[string]schema.Attribute{
+					"public_endpoint": schema.SingleNestedAttribute{
+						MarkdownDescription: "The public endpoint for connecting to the cluster.",
+						Computed:            true,
+						Attributes: map[string]schema.Attribute{
+							"host": schema.StringAttribute{
+								MarkdownDescription: "The host of the public endpoint.",
+								Computed:            true,
+							},
+							"port": schema.Int64Attribute{
+								MarkdownDescription: "The port of the public endpoint.",
+								Computed:            true,
+							},
+							"disabled": schema.BoolAttribute{
+								MarkdownDescription: "Whether the public endpoint is disabled.",
+								Computed:            true,
+							},
+						},
+					},
+					"private_endpoint": schema.SingleNestedAttribute{
+						MarkdownDescription: "The private endpoint for connecting to the cluster.",
+						Computed:            true,
+						Attributes: map[string]schema.Attribute{
+							"host": schema.StringAttribute{
+								MarkdownDescription: "The host of the private endpoint.",
+								Computed:            true,
+							},
+							"port": schema.Int64Attribute{
+								MarkdownDescription: "The port of the private endpoint.",
+								Computed:            true,
+							},
+							"aws_endpoint": schema.SingleNestedAttribute{
+								MarkdownDescription: "Message for AWS PrivateLink information.",
+								Computed:            true,
+								Attributes: map[string]schema.Attribute{
+									"service_name": schema.StringAttribute{
+										MarkdownDescription: "The AWS service name for private access.",
+										Computed:            true,
+									},
```

Comment:
why not follow api name, just `public`/`private`/`aws`, and remove `gcp` plz, since we don't have it.

Reply:
Cause I think these names may misunderstand users, it's ok to use public/private/aws